src/login/AzureSessionInternal.ts

```diff
@@ -3,7 +3,26 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AccountInfo } from "@azure/msal-common";
+import { Environment } from '@azure/ms-rest-azure-env';
+import { AccountInfo } from '@azure/msal-common';
 import { AzureSession } from "../azure-account.api";
+import { localize } from '../utils/localize';
+import { AbstractCredentials, AbstractCredentials2 } from './AuthProviderBase';
 
-export type AzureSessionInternal = AzureSession & { readonly accountInfo?: AccountInfo };
+export class AzureSessionInternal implements AzureSession {
+	constructor(
+		public environment: Environment,
+		public userId: string,
+		public tenantId: string,
+		public accountInfo: AccountInfo | undefined,
+		private _credentials: AbstractCredentials | undefined,
+		public credentials2: AbstractCredentials2,
+	) {}
+
+	public get credentials(): AbstractCredentials {
```

Comment:
Not quite - we want the MSAL error inside the MsalAuthProvider, not in a generic class like this.

You need to delay calling `getCredentials` until it's actually used. Something like this:
```typescript
constructor(
    ...
    private _authProvider: AuthProviderBase<unknown>,
) { }

public get credentials(): AbstractCredentials {
    return this._authProvider.getCredentials(this.environment.name, this.userId, this.tenantId);
}

public get credentials2(): AbstractCredentials2 {
    return this._authProvider.getCredentials2(this.environment, this.userId, this.tenantId, this.accountInfo);
}
```